components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java

```diff
@@ -182,40 +192,121 @@ public Secret replaceSecret(Secret secret) throws SecretManagementException {
+        }
+        return secretType;
+    }
+
+    @Override
+    public void deleteSecretType(String secretTypeName) throws SecretManagementException {
+
+        validateSecretTypeDeleteRequest(secretTypeName);
+        getSecretDAO().deleteSecretTypeByName(secretTypeName);
+
+        if (log.isDebugEnabled()) {
+            log.debug("Secret type: " + secretTypeName + " is successfully deleted.");
+        }
+    }
+
     /**
-     * Validate that secret name is non-empty.
+     * Validate that secret type and secret name is non-empty.
      *
+     * @param secretTypeName Name of the {@link SecretType}.
      * @param secretName The secret name.
      * @throws SecretManagementException If secret validation fails.
      */
-    private void validateSecretRetrieveRequest(String secretName) throws SecretManagementException {
+    private void validateSecretRetrieveRequest(String secretTypeName, String secretName) throws SecretManagementException {
+
+        if (StringUtils.isEmpty(secretTypeName) || StringUtils.isEmpty(secretName)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Invalid secret identifier with secretName: " + secretName
+                        + " and secretTypeName: " + secretName + ".");
+            }
+            throw handleClientException(ERROR_CODE_SECRET_GET_REQUEST_INVALID, null);
+        }
+    }
+
+    private void validateSecretsRetrieveRequest(String secretTypeName)
+            throws SecretManagementException {
 
-        if (StringUtils.isEmpty(secretName)) {
+        if (StringUtils.isEmpty(secretTypeName)) {
             if (log.isDebugEnabled()) {
-                log.debug("Invalid secret identifier with secretName: " + secretName + ".");
+                log.debug("Invalid secret identifier with secretTypeName: " + secretTypeName + ".");
             }
             throw handleClientException(ERROR_CODE_SECRET_GET_REQUEST_INVALID, null);
         }
     }
 
     /**
-     * Validate that secret name is non-empty.
+     * Validate that secret type and secret name is non-empty.
      * Set tenant domain if they are not set to the secret object.
      *
+     * @param secretTypeName Name of the {@link SecretType}.
      * @param secretName The secret name.
      * @throws SecretManagementException If secret validation fails.
      */
-    private void validateSecretDeleteRequest(String secretName)
+    private void validateSecretDeleteRequest(String secretTypeName, String secretName)
             throws SecretManagementException {
 
-        if (StringUtils.isEmpty(secretName)) {
+        if (StringUtils.isEmpty(secretTypeName) || StringUtils.isEmpty(secretName)) {
             if (log.isDebugEnabled()) {
-                log.debug("Error identifying the secret with secret name: " + secretName + ".");
+                log.debug("Error identifying the secret with secret name: " + secretName + " and secret type:"
```

Comment:
`log.debug("Error identifying the secret with secret name: " + secretName + " and secret type: "`

Reply:
Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)